src/main/kotlin/api/Door43Retrieval.kt

```diff
@@ -0,0 +1,23 @@
+package api
+
+import api.model.Door43Mapper
+import io.reactivex.Observable
+import persistence.injection.DaggerDatabaseComponent
+
+class Door43Retrieval(val retrofitConfig: RetrofitConfig = RetrofitConfig()) {
+
+    val appDatabase = DaggerDatabaseComponent.builder().build().inject()
+    val languageDao = appDatabase.getLanguageDao()
```

Comment:
the actual retrieval and insertion of languages should be done through a use case, this is a violation of the single responsibility principle. Similar to your AppDatabase, you want to be creating a Door43Client that exposes various functionality such as getting languages.

Reply:
@sarabiaj So create a use case that handles getting languages and putting them in the database, and create an IDoor43Client interface that is implemented by Door43Retrieval (which has the retrofit specifics). Inside the use case, we inject the IDoor43Client (which is implemented by Door43Retrieval), and then use it to get the languages and put them in the database (also injected into the use case)

Reply:
yeah, I would use the word Client instead of Retrieval though. I'm fine with IClassName being the interface and ClassName being the implementation, if you don't want ClassNameImpl.